### PR TITLE
Update test_numpy_array

### PR DESCRIPTION
**Pull request by jgersak**
rename test_numpy_array into assert_statistics_match update tests
add option for absolute delta
add test assert_statistics_match
add deprecation warning

### CODE REVIEW

File: sentinelhub/testing_utils.py [hunk 1]
```diff
@@ -28,6 +29,23 @@ def test_numpy_data(
...
     exp_std: Optional[float] = None,
     delta: Optional[float] = None,
+) -> None:
+    """Deprecated version of assert_statistics_match"""
+    warn('Deprecated, rather use function "assert_statistics_match" look changelog', DeprecationWarning, stacklevel=2)
+    assert_statistics_match(data, exp_shape, exp_dtype, exp_min, exp_max, exp_mean, exp_median, exp_std, delta)
+
+
+def assert_statistics_match(
+    data: Optional[np.ndarray] = None,
```
[the remark this answers is not in the document]
Comment (jgersak) [reply sentinelhub/testing_utils.py:39]: the org `test_numpy_array`had it i keep it "the same" but i m going to remove it

File: tests/test_testing_utils.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                "exp_mean": 50,
+                "exp_median": 50,
+                "delta": 0.05,
+                "abs_delta": True,
+            },
+            False,
+        ],
+    ),
+)
+def test_assert_statistics_match(data: np.array, expected_statistics: Dict, succeed: bool) -> None:
```
[the remark this answers is not in the document]
Comment (jgersak) [reply tests/test_testing_utils.py]: done

File: sentinelhub/testing_utils.py [hunk 3]
```diff
@@ -61,6 +80,12 @@ def test_numpy_data(
...
         if expected is not None:
             data_stats[name] = func(data)  # type: ignore # unknown function
-            exp_stats[name] = expected if name in is_precise else approx(expected, rel=delta)
+            exp_stats[name] = (
+                expected
+                if name in is_precise
+                else approx(expected, rel=delta)
+                if not abs_delta
+                else approx(expected, abs=delta)
+            )
```
[the remark this answers is not in the document]
Comment (jgersak) [reply sentinelhub/testing_utils.py]: updeted

File: sentinelhub/testing_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jgersak) [reply sentinelhub/testing_utils.py]: done

File: sentinelhub/testing_utils.py [hunk 4]
```diff
@@ -28,6 +29,23 @@ def test_numpy_data(
     exp_median: Optional[float] = None,
     exp_std: Optional[float] = None,
     delta: Optional[float] = None,
+) -> None:
+    """Deprecated version of assert_statistics_match"""
+    warn('Deprecated, rather use function "assert_statistics_match" look changelog', DeprecationWarning, stacklevel=2)
```
[the remark this answers is not in the document]
Comment (jgersak) [reply sentinelhub/testing_utils.py]: done

File: sentinelhub/testing_utils.py [hunk 5]
```diff
@@ -28,6 +29,23 @@ def test_numpy_data(
...
+    data: Optional[np.ndarray] = None,
+    exp_shape: Optional[Tuple[int, ...]] = None,
+    exp_dtype: Optional[np.dtype] = None,
+    exp_min: Optional[float] = None,
+    exp_max: Optional[float] = None,
+    exp_mean: Optional[float] = None,
+    exp_median: Optional[float] = None,
+    exp_std: Optional[float] = None,
+    delta: Optional[float] = None,
+    abs_delta: Optional[bool] = False,
```
[the remark this answers is not in the document]
Comment (jgersak) [reply sentinelhub/testing_utils.py]: updated

**Review by zigaLuksic — APPROVED** [review]

**Pull request merged**